tests/test_sequence_block.py

```diff
@@ -162,3 +162,42 @@ def test_nested():
 """)
     stream[0].extend(("c", "d"))
     assert stream.text == b'-\n - a\n - b\n - "c"\n - "d"\n'
+
+def test_same_line_nested():
+    stream = parse_str("""\
+- - x
+  - y
+""")
+    stream[0][0] = "z"
+    assert stream.text == b'''\
+-
+  - "z"
+  - y
+'''
+
+def test_same_line_triple_nested():
+    stream = parse_str("""\
+- - - x
+    - y
+  - - z
+""")
+    stream[0][0][0] = "mmm"
+    assert stream.text == b'''\
+- -
+    - "mmm"
+    - y
+  - - z
+'''
```

Comment:
Interesting. I would have thought this would become: 

```
-
  -
    - "mmm"
    - y
     - - z
```

But thinking about this a bit more I think I understand why it behaves the way it does. It tries to fix (or emulate) the closest item and in this case it is the inner most sequence.

Reply:
Right.  The inconsistency comes from the (used to be in the readme) goal of "only edit modified lines".  It doesn't _need_ to change the first-line style of the outer and middle sequence, so they are left alone.

If you did something to the middle list like a slice assignment, it would anneal that (and the inner list by extension) because it means 1/10th as much code to do that, and most of my use is editing leaf nodes anyway.